capstone/src/main/java/com/google/sps/data/AndroidManifestParser.java

```diff
@@ -1,9 +1,182 @@
+        int tag0 = LEW(xml, off);
+        //int tag1 = LEW(xml, off+1*4);
+        int lineNo = LEW(xml, off+2*4);
+        //int tag3 = LEW(xml, off+3*4);
+        int nameNsSi = LEW(xml, off+4*4);
+        int nameSi = LEW(xml, off+5*4);
+    
+        if (tag0 == startTag) { // XML START TAG
+            int tag6 = LEW(xml, off+6*4);  // Expected to be 14001400
+            int numbAttrs = LEW(xml, off+7*4);  // Number of Attributes to follow
+            //int tag8 = LEW(xml, off+8*4);  // Expected to be 00000000
+            off += 9*4;  // Skip over 6+3 words of startTag data
+            String name = compXmlString(xml, sitOff, stOff, nameSi);
+            //tr.addSelect(name, null);
+            startTagLineNo = lineNo;
+        
+            // Look for the Attributes
+            StringBuffer sb = new StringBuffer();
+            for (int ii=0; ii<numbAttrs; ii++) {
+                int attrNameNsSi = LEW(xml, off);  // AttrName Namespace Str Ind, or FFFFFFFF
+                int attrNameSi = LEW(xml, off+1*4);  // AttrName String Index
+                int attrValueSi = LEW(xml, off+2*4); // AttrValue Str Ind, or FFFFFFFF
+                int attrFlags = LEW(xml, off+3*4);  
+                int attrResId = LEW(xml, off+4*4);  // AttrValue ResourceId or dup AttrValue StrInd
+                off += 5*4;  // Skip over the 5 words of an attribute
+            
+                String attrName = compXmlString(xml, sitOff, stOff, attrNameSi);
+                String attrValue = attrValueSi!=-1
+                    ? compXmlString(xml, sitOff, stOff, attrValueSi)
+                    : "resourceID 0x"+Integer.toHexString(attrResId);
+                sb.append(" "+attrName+"=\""+attrValue+"\"");
+                getPermissions(attrValue);
+            }
+            //prtIndent(indent, "<"+name+sb+">");
+            indent++;
+        
+        } else if (tag0 == endTag) { // XML END TAG
+            indent--;
+            off += 6*4;  // Skip over 6 words of endTag data
+            String name = compXmlString(xml, sitOff, stOff, nameSi);
+            //prtIndent(indent, "</"+name+">  (line "+startTagLineNo+"-"+lineNo+")");
+            //tr.parent();  // Step back up the NobTree
+        
+        } else if (tag0 == endDocTag) {  // END OF XML DOC TAG
+            break;
+        
+        } else {
+            // System.out.println("  Unrecognized tag code '"+Integer.toHexString(tag0)
+            // +"' at offset "+off);
+            break;
+            }
+    } // end of while loop scanning tags and attributes of XML tree
+        //System.out.println("    end at offset "+off);
+    } // end of decompressXML
+        
+    
+    public String compXmlString(byte[] xml, int sitOff, int stOff, int strInd) {
+        if (strInd < 0) 
+            return null;
+        int strOff = stOff + LEW(xml, sitOff+strInd*4);
+    return compXmlStringAt(xml, strOff);
+    }   
+    
+    public void prtIndent(int indent, String str) {
```

Comment:
Either remove this or add a docstring describing how this is useful in debugging.